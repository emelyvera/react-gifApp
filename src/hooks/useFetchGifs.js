import { useState, useEffect } from "react";
import { getGifs } from '../helpers/getGifs';

export const useFetchGifs = (category) => {
  const [objeto, setObjeto] = useState({
    data: [],
    loading: true
  });

  useEffect(() => {

    getGifs(category)
      .then(imgs => {

        setObjeto({
          data: imgs,
          loading: false
        })


      });


  }, [category])


  return objeto;

};
