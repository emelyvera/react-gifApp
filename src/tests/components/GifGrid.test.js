import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs')

describe('Pruebas del componente <GifGrid />', () => {
    
    const category = 'Miedo'
  
  test('should de mostrar el componente correctamete <GifGrid />', () => {

    useFetchGifs.mockReturnValue({
        data: [],
        loading: true
    });
    
    const wrapper = shallow( <GifGrid category={category}/>);
    expect( wrapper ).toMatchSnapshot();
    
  });

  test('should  de mostrar items cuando se cargan img useFetchGifs', () => {
    
    const gifs = [{
        id:'ABC',
        url: 'https://local.com',
        title: 'holaMundo'
    }]; 

    useFetchGifs.mockReturnValue({
        data: gifs,
        loading: false
    });

    const wrapper = shallow( <GifGrid category={category}/>);

   expect( wrapper.find('p').exists() ).toBe(false)
   expect( wrapper.find('GifGridItem').length ).toBe( gifs.length )


  });
  
    
});
