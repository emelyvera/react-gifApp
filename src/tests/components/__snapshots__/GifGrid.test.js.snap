// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente <GifGrid /> should de mostrar el componente correctamete <GifGrid /> 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeInDown"
  >
    Miedo
  </h3>
  <p
    className="animate__heartBeat"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
